test/testScan.js

```diff
@@ -62,6 +62,29 @@ builder.add(function testScanSegment(test) {
   return scanAndCheck(scan, [1, 3, 4], test)
 })
 
+builder.add(function testScanOnGlobalSecondaryIndex(test) {
+  var scan = this.client.newScanBuilder(tableName)
+               .setIndexNameGenerator(utils.indexNameGenerator)
+               .setHashKey('post')
+               .setRangeKey('email')
+  return scanAndCheck(scan, [0, 2, 5], test)
+})
+
+builder.add(function testScanOnGlobalSecondaryIndexWithoutRangeKey(test) {
+  var scan = this.client.newScanBuilder(tableName)
+               .setIndexNameGenerator(utils.indexNameGenerator)
+               .setHashKey('description')
+  return scanAndCheck(scan, [5], test)
+})
+
+builder.add(function testParallelScanOnGlobalSecondaryIndex(test) {
+  var scan = this.client.newScanBuilder(tableName)
+               .setIndexNameGenerator(utils.indexNameGenerator)
+               .setHashKey('post')
+               .setParallelScan(1, 2)
+  return scanAndCheck(scan, [0, 2, 5, 6], test)
```

Comment:
i'm pretty confident this is deterministic, yes. if it wasn't, the segmentation strategy that amazon used could lead to overlaps or missed rows.